### PR TITLE
Beta Release 30 - Includes up to Developer Release 5817

### PR DESCRIPTION
**Pull request by cozza13**
http://rawgit.com/cozza13/QAtesting/master/QAofRC.js

* Threaded Audio. Adds multithreading support for the audio-mixer assignment-client.
* Photobooth MVP - Dev tool for taking standardized product pictures ( https://github.com/highfidelity/hifi/pull/9174 )
* Exposing last edited property of Entity to Javascript ( https://github.com/highfidelity/hifi/pull/9164) 
* Honor reorient and quit request from OVR 
* Update Xbox controller mapping to have the menu on Start 
* First-order Ambisonic renderer ( https://github.com/highfidelity/hifi/pull/9125 )
* Add account sign up to Interface 
* Restores "Select Audio DLL at runtime" 
* avatarFinderBeacon.js -- Find other avatars more easily. This is a script that will add a red line from 1km above till 1km below the avatars out side of a 20 meter radius. 
* Add test mode to interface 
* Add support for oculus touch in help menu 
* Use smart pointers to avoid leaking EntityTreeElementExtraEncodeData
* Fix and improve the WAV file loader 
* Bubble visualization improvements ( https://github.com/highfidelity/hifi/pull/9199 )
* Kinect Plugin: Implements support for hand poses from kinect camera ( https://github.com/highfidelity/hifi/pull/9186 )
* Drop avatar data rate to 45hz & bump default per client bandwidth from mixer 
* Fix Oculus Touch flags check 
* Fix local light shading tiling issue 
* Make Steam client calls into a plugin 
* Fix a confusing comment about min cores required
* Use JSON for application Settings: Given the way the ini "file format" (There is no actually specified format) works and the way Qt decided to interpret it means that integer value and boolean don't retain their type past the life of the process. Any subsequent read will return a String. 
* Fix for Gif snapshots crashing interface
* Adds a forward render task to the graphics engine 
* Change file permissions of source code to remove execute bits
* Rescale yourself or entities using the grip buttons ( https://github.com/highfidelity/hifi/pull/9187 ) 
* People Action List prototype: Creates a People Action List with the same functionality as the mod.js
* Ambisonic resampler, optimized using AVX2 
* Adjust positioning of avatar hands to better match touch controller position 
* Add profile stats relating to avatar CPU costs 
* Render bounds in Forward Render Task 
* No longer applying volume when reading audio injector buffer 
* Add touch support to tutorial 
* Additional work on tracing

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: ee3c8e6efd6033e6630b08dfa5563255687eb0af available at: https://gist.github.com/25418380d87274ace1d3d37e9a60f66f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9230-ee3c8e6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9230-ee3c8e6.dmg

**Comment by kevinmthomasse** [comment]
testing

**Comment by Nex-Pro** [comment]
Testing in progress.

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: KevinMThomas
**localhost 'createTestBlocks.js' Result**: **localhost 'createTestBlocks.js'** SUCCESS
**localhost ATP Upload + Edit Result**: **localhost ATP** SUCCESS
**Echo Server Audio**: **Echo Server Audio** SUCCESS
**localhost 'harmonicOscillator.js' Result**: **localhost 'harmonicOscillator.js'** NOT TESTED
**localhost 'reverbTest.js' Result**: **localhost 'reverbTest.js'** SUCCESS
**localhost 'playaPerformanceTest.js' Result**: **localhost 'playaPerformanceTest.js'** SUCCESS
**dev-playa 'playaPerformanceTest.js' Result**: **dev-playa 'playaPerformanceTest.js'** SUCCESS
**dev-demo 'playaPerformanceTest.js' Result**: **dev-demo 'playaPerformanceTest.js'** SUCCESS
**QA Pass**: **QA Pass** FAILURE
**Additional Test Results + Observations**: Failure as switching between the ```KevinMThomas Fuse Avatar``` and ```Michael``` and ```Priscialla``` were expected behavior however when switching back to the ```KevinMThomas Fuse Avatar``` a crash ensued in addition content and avatars took about 2 minutes longer than usual to rez.  Unable to replicate the crash as I tried twice more however this should not have occurred even once on a RC release in my opinion.  If other testers read this PLEASE TEST FOR THIS and report results to compare.  Left Oculus Touch controller was NOT active and was not working properly.  I did a controlled test with the Steam release and both controllers DID work properly to see if it was my setup to which it was not.


**Domain** |localhost |dev-playa |dev-demo
---|---|---|---
**Render Rate**: |60 |60 |57
**Present Rate**: |60 |60 |57
**Present New Rate**: |54 |60 |43
**CPU Buffer Count**: |1994 |855 |705
**GPU Buffer Count**: |671 |199 |236
**CPU Memory Usage**: |19850512 |7725232 |7761744
**GPU Memory Usage**: |13243824 |5973424 |6137264
**CPU Texture Count**: |232 |87 |84
**GPU Texture Count**: |190 |70 |66
**GPU Transfer Count**: |0 |0 |0
**CPU Texture Memory Usage**: |121744560 |49614096 |49614096
**GPU Texture Memory Usage**: |614937873 |275824555 |258031531
**GPU Virtual Memory Usage**: |601442415|271471715|254760092
**Triangle Count**: |245046 |70955 |160618
**Triangle Rate**: |14414470 |4173823 |10038625
**API Drawcall Count**: |185 |54 |74
**Drawcall Count**: |2996 |51 |74
**Drawcall Rate**: |176235 |3000 |4625
**State Changes Texture Count**:|353 |79 |107
**Pipeline Count**: |68 |33 |47
**Item Opaques**: |260 |210 |88
**Item Translucents**: |89 |46 |93
**Item Lights**: |11 |40 |43
**Timing Opaques**: |1.367 |1.08 |0.281
**Timing Translucents**: |0.265 |0.051 |0.369
**Timing Lighting**: |0.019 |0.015 |0.015
**Render Frame**: |2.871 |2.342 |1.829


**RC Specific tests**

**Test** | **Description** | **Results**
---|---|---
**Test 1**|For performance testing use. Localhost - dev-playa - dev-demo|Success
**Test 2**|Install a fresh copy of "Home' and do tutorial|Did not prompt me to do so - not a failure just did not prompt as it normally does.
**Test 3**|Go to localhost - Make sure you can hear the background sounds.|Success
**Test 4**|Launch firework, on dev-playa, can other hear?|Success
**Test 5**|Switch avatar and make sure it does not crash|Success - it did take a about 2 minutes more than usual for ```Matthew``` and ```Priscialla``` to load however successful.
**Test 6**|Test group functions|Untested
**Test 7**|Test Kick now includes MAC address|Untested
**Test 8**|Enrure that you can distance grab|Failure with Oculus Touch ONLY on right controller left controller did not work.  Tested on steam in current build as a comparison and both DID work  This is a Failure.
**Test 9**|Bring up and edit an objects with hand controllers|Failure with Oculus Touch ONLY on right controller left controller did not work.  Tested on steam in current build as a comparison and both DID work  This is a Failure.
**Test 10**|Grab objects in dev-demo|Failure with Oculus Touch ONLY on right controller left controller did not work.  Tested on steam in current build as a comparison and both DID work  This is a Failure.
**Test 11**|Test ignore function|Untested

**Comment by OmegaHeron** [comment]
Windows 10 - i5 6500 CPU - GTX970 GPU

## QA Test
**QA Tester**: OmegaHeron
**localhost 'createTestBlocks.js' Result**: **localhost 'createTestBlocks.js'** SUCCESS
**localhost ATP Upload + Edit Result**: **localhost ATP** SUCCESS
**Echo Server Audio**: **Echo Server Audio** SUCCESS
**localhost 'harmonicOscillator.js' Result**: **localhost 'harmonicOscillator.js'** SUCCESS
**localhost 'reverbTest.js' Result**: **localhost 'reverbTest.js'** SUCCESS
**localhost 'playaPerformanceTest.js' Result**: **localhost 'playaPerformanceTest.js'** SUCCESS
**dev-playa 'playaPerformanceTest.js' Result**: **dev-playa 'playaPerformanceTest.js'** SUCCESS
**dev-demo 'playaPerformanceTest.js' Result**: **dev-demo 'playaPerformanceTest.js'** SUCCESS
**QA Pass**: **QA Pass** SUCCESS
**Additional Test Results + Observations**:

**RC Specific tests**

**Test** | **Description** | **Results**
---|---|---
**Test 1**|For performance testing use. Localhost - dev-playa - dev-demo|Yes
**Test 2**|Install a fresh copy of "Home' and do tutorial|Yes
**Test 3**|Go to localhost - Make sure you can hear the background sounds.|Pass
**Test 4**|Launch firework, on dev-playa, can other hear?|N/A
**Test 5**|Switch avatar and make sure it does not crash|Pass
**Test 6**|Test group functions|Pass
**Test 7**|Test Kick now includes MAC address|Pass
**Test 8**|Ensure that you can distance grab|N/A
**Test 9**|Bring up and edit an objects with hand controllers|N/A
**Test 10**|Grab objects in dev-demo|Pass
**Test 11**|Test ignore function|Pass

Desktop 1600x900 Maximized Window

**Domain** |localhost |dev-playa |dev-demo
---|---|---|---
**Render Rate**: |358 |264 |276
**Present Rate**: |357 |264 |276
**Present New Rate**: |270 |153 |236
**CPU Buffer Count**: |1741 |4467 |3814
**GPU Buffer Count**: |634 |1399 |982
**CPU Memory Usage**: |16048592 |71906160 |65869808
**GPU Memory Usage**: |10687088 |47556720 |32558448
**CPU Texture Count**: |222 |296 |320
**GPU Texture Count**: |192 |253 |291
**GPU Transfer Count**: |0 |0 |0
**CPU Texture Memory Usage**: |81766768 |175443216 |97450448
**GPU Texture Memory Usage**: |575909373 |1125447918 |1124563368
**GPU Virtual Memory Usage**: |562364779|1114350113|1153928340
**Triangle Count**: |237796 |869941 |358920
**Triangle Rate**: |118898000 |173988200 |119640000
**API Drawcall Count**: |184 |292 |199
**Drawcall Count**: |2997 |335 |1105
**Drawcall Rate**: |1498500 |67000 |368333
**State Changes Texture Count**:|369 |347 |315
**Pipeline Count**: |58 |34 |52
**Item Opaques**: |267 |475 |226
**Item Translucents**: |60 |55 |77
**Item Lights**: |11 |43 |41
**Timing Opaques**: |0.959 |2.158 |0.75
**Timing Translucents**: |0.141 |0.102 |0.199
**Timing Lighting**: |0.013 |0.03 |0.022
**Render Frame**: |1.871 |4.614 |2.286

DK2

**Domain** |localhost |dev-playa |dev-demo
---|---|---|---
**Render Rate**: |75 |75 |75
**Present Rate**: |75 |75 |75
**Present New Rate**: |75 |75 |75
**CPU Buffer Count**: |2560 |4691 |3156
**GPU Buffer Count**: |840 |1415 |857
**CPU Memory Usage**: |33721168 |73286512 |49463824
**GPU Memory Usage**: |22452336 |47994992 |26658928
**CPU Texture Count**: |236 |345 |295
**GPU Texture Count**: |206 |304 |248
**GPU Transfer Count**: |0 |0 |0
**CPU Texture Memory Usage**: |50850992 |215944464 |218230032
**GPU Texture Memory Usage**: |813426585 |1301209651 |1036022593
**GPU Virtual Memory Usage**: |799193835|1453542078|1021664373
**Triangle Count**: |488693 |1739890 |814677
**Triangle Rate**: |37591769 |133837692 |62667461
**API Drawcall Count**: |190 |292 |225
**Drawcall Count**: |3658 |658 |19096
**Drawcall Rate**: |281384 |50615 |1468923
**State Changes Texture Count**:|374 |358 |330
**Pipeline Count**: |60 |33 |56
**Item Opaques**: |266 |480 |254
**Item Translucents**: |82 |56 |80
**Item Lights**: |11 |44 |44
**Timing Opaques**: |1.193 |1.892 |0.956
**Timing Translucents**: |0.159 |0.144 |0.16
**Timing Lighting**: |0.023 |0.028 |0.025
**Render Frame**: |2.664 |4.29 |2.761

I switched between my normal mixamo fuse AV, being of light and a several models from MP without issue here.

Pass for all specific items below;

Checked PAL functions
Checked CPU use (detail graphs to follow, but, all good)
Checked ambisonic sound
Checked avatar data rate change
Checked ini format to json format settings migration/persistence
Checked fix for Gif snapshots crashing interface

**Comment by Nex-Pro** [comment]
### Beta Release 30 - Includes up to Developer Release 5817

### Please read notes:

#### #9230

Testing platform:

CPU: i7 4ghz
RAM: 16gb
GPU: Nvidia GTX 1070
OS : Win10
HMD: HTC Vive
Hand Controllers: HTC Vive

Crash on exit via toolbar icon. No splat generated.

GPU load does not reduce when Interface is minimized: Posted to FogBugz.

## QA Test

**QA Tester**: ***Piper.Peppercorn***
**localhost 'createTestBlocks.js' Result**: ***SUCCESS***
**localhost ATP Upload + Edit Result**: ***SUCCESS***
**Echo Server Audio**: ***SUCCESS***
**localhost 'harmonicOscillator.js' Result**: ***FAIL***
**localhost 'reverbTest.js' Result**: ***SUCCESS***
**localhost 'playaPerformanceTest.js' Result**: ***UNTESTED***
**dev-playa 'playaPerformanceTest.js' Result**: ***UNTESTED***
**dev-demo 'playaPerformanceTest.js' Result**: ***UNTESTED***

**QA Pass: **PASS** ***Please read notes***

**Additional Test Results + Observations**:

Harmonic oscillator on my sandbox was very unreliable. Stuttering. Worst I've seen it in a long time. Other avatar ( not local ) Also reported poor performance for this test.
Several crashes on exit. No splats generated.


**Domain** |localhost |dev-playa |dev-demo
---|---|---|---
**Render Rate**: |90 |90 |90
**Present Rate**: |89 |90 |90
**Present New Rate**: |90 |57 |64
**CPU Buffer Count**: |3670 |5063 |4523
**GPU Buffer Count**: |901 |1557 |1381
**CPU Memory Usage**: |53623632 |90067824 |94652400
**GPU Memory Usage**: |32847984 |65591408 |62991728
**CPU Texture Count**: |270 |307 |359
**GPU Texture Count**: |214 |269 |326
**GPU Transfer Count**: |0 |0 |0
**CPU Texture Memory Usage**: |72248496 |162401552 |52759824
**GPU Texture Memory Usage**: |1018870677 |1384478255 |1393018683
**GPU Virtual Memory Usage**: |1243221709|1766476251|1864926356
**Triangle Count**: |800355 |1471695 |1717088
**Triangle Rate**: |72759545 |133790454 |156098909
**API Drawcall Count**: |194 |116 |277
**Drawcall Count**: |6000 |218 |2346
**Drawcall Rate**: |545454 |19818 |213272
**State Changes Texture Count**:|384 |165 |390
**Pipeline Count**: |62 |33 |52
**Item Opaques**: |276 |182 |312
**Item Translucents**: |107 |25 |73
**Item Lights**: |11 |30 |39
**Timing Opaques**: |1.153 |0.661 |0.998
**Timing Translucents**: |0.21 |0.041 |0.129
**Timing Lighting**: |0.022 |0.019 |0.022
**Render Frame**: |2.385 |1.99 |2.217


**RC Specific tests**

**Test** | **Description** | **Results**
---|---|---
**Test 1**|For performance testing use. Localhost - dev-playa - dev-demo| **CHECK**
**Test 2**|Install a fresh copy of "Home' and do tutorial|  **SUCCESS**
**Test 3**|Go to localhost - Make sure you can hear the background sounds.| **SUCCESS**
**Test 4**|Launch firework, on dev-playa, can other hear?| **SUCCESS**
**Test 5**|Switch avatar and make sure it does not crash| **SUCCESS**
**Test 6**|Test group functions| **BASIC TESTS SUCCESSFUL**
**Test 7**|Test Kick now includes MAC address| **SUCCESS**
**Test 8**|Enrure that you can distance grab| **SUCCESS**
**Test 9**|Bring up and edit an objects with hand controllers| **FAIL**
**Test 10**|Grab objects in dev-demo|undefined
**Test 11**|Test ignore function| **SUCCESS**


## NOTES:

### Edit an object while in HMD:

Unble to enter any user data as no QML keyboard is raised.

### Ambisonic:

I could hear all three test wavs via script running in my interface and via entitly placed in my sandbox.
Michelle was unable to hear the ambisonic wav via my interface but could hear via entity. We tested this several times.
She could hear the mono and stereo examples via entity or my interface.

### Photobooth:

Photobooth cannot be used in HMD mode.
Avatar must be several feet above local terrain/floor or booth will rez too low.
Right click to paste modelURL raises context popup that cannot be dismissed. The user's view then goes to 0,0,0 ( if a camera view other than first person was previously selected )
User then has to select first person view. At this point the studio is no longer in existence.( at dev-welcome, might be an issue there with entity lifetime.)
Imported model is rescaled and moved below the backdrop.
Some models appear to have had their dimensions reset resulting in a tiny object.
Photobooth assumes all models are oriented the same way. Adding some simple rotate buttons would make sense for a faster workflow.

### Avatar scale with gripping:

Try scaling avatar whilst inside the castle at welcome.
Basically the side effect of shooting laser/s whilst rescaling is unwanted and cumbersome.
This might be better toggled via a gesture or option.

### Entity scale with grip:

Avatar with no edit perms can rescale grabbable objects.
Generally ok apart from object scales from center ( maybe move registration point to hand location for near grab while scaling)
Scaling is increasingly granular as object size is reduced making for an unsatisfactory experience with small objects.
Scaling should be proportional rather than incremental.

### Bugsplat:

Bugsplats are not triggered by this build.

## All I wanna do is shrink meh !

r3

### Michelle is so strong she can overcome no-edit perms !
hifi-snap-by-piper-peppercorn-on-2016-12-17_17-49-05

**Comment by Nex-Pro** [comment]
Additionally There were some strange lighting issues. Lighting on objects changing dramatically with head angle.
hifi-snap-by-piper-peppercorn-on-2016-12-17_20-15-55

Voxel paint objects become degraded by leaving and returning to domain. The degradation changes with each re-visit.
hifi-snap-by-piper-peppercorn-on-2016-12-17_20-32-00

here the zone lighting did not load.
hifi-snap-by-piper-peppercorn-on-2016-12-17_20-33-34

**Comment by claudioshg** [comment]
## QA Test
**QA Tester**: claudioshg
**localhost 'createTestBlocks.js' Result**: **localhost 'createTestBlocks.js'** SUCCESS
**localhost ATP Upload + Edit Result**: **localhost ATP** SUCCESS
**Echo Server Audio**: **Echo Server Audio** SUCCESS
**localhost 'harmonicOscillator.js' Result**: **localhost 'harmonicOscillator.js'** SUCCESS
**localhost 'reverbTest.js' Result**: **localhost 'reverbTest.js'** SUCCESS
**localhost 'playaPerformanceTest.js' Result**: **localhost 'playaPerformanceTest.js'** SUCCESS
**dev-playa 'playaPerformanceTest.js' Result**: **dev-playa 'playaPerformanceTest.js'** SUCCESS
**dev-demo 'playaPerformanceTest.js' Result**: **dev-demo 'playaPerformanceTest.js'** SUCCESS
**QA Pass**: **QA Pass** SUCCESS
**Additional Test Results + Observations**:


**Domain** |localhost |dev-playa |
---|---|---|---
**Render Rate**: |0 |0 |
**Present Rate**: |0 |0 |
**Present New Rate**: |0 |0 |
**CPU Buffer Count**: |2269 |4150 |
**GPU Buffer Count**: |787 |1093 |
**CPU Memory Usage**: |26680016 |70741488 |
**GPU Memory Usage**: |18348016 |41280240 |
**CPU Texture Count**: |212 |270 |
**GPU Texture Count**: |184 |202 |
**GPU Transfer Count**: |0 |0 |
**CPU Texture Memory Usage**: |46460080 |252644624 |
**GPU Texture Memory Usage**: |859450489 |1160226574 |
**GPU Virtual Memory Usage**: |845523566|1150221038|
**Triangle Count**: |75174 |2356827 |
**Triangle Rate**: |7517400 |235682700 |
**API Drawcall Count**: |31 |286 |
**Drawcall Count**: |5322 |644 |
**Drawcall Rate**: |532200 |64400 |
**State Changes Texture Count**:|58 |329 |
**Pipeline Count**: |31 |42 |
**Item Opaques**: |21 |483 |
**Item Translucents**: |9 |73 |
**Item Lights**: |1 |50 |
**Timing Opaques**: |0.108 |1.361 |
**Timing Translucents**: |0.025 |0.084 |
**Timing Lighting**: |0.027 |0.024 |
**Render Frame**: |0.994 |3.478 |



**RC Specific tests**

**Test** | **Description** | **Results**
---|---|---
**Test 1**|For performance testing use. Localhost - dev-playa - dev-demo|In Dev-Demo, there is an area between the whiteboard and the tether ball, that while the avatar stands in it, the sky flickers from black to the original image. Looking left and right consistantly triggers the flickering. In Dev-Playa, when both "Launch" buttons are pressed in rapid succession, the entire room resets right after the fireworks have gone off.
**Test 2**|Install a fresh copy of "Home' and do tutorial|Untested
**Test 3**|Go to localhost - Make sure you can hear the background sounds.|Success
**Test 4**|Launch firework, on dev-playa, can other hear?|Success - Launching too many fireworks at once crashes the room and resets.
**Test 5**|Switch avatar and make sure it does not crash|Success
**Test 6**|Test group functions|Untested
**Test 7**|Test Kick now includes MAC address|Untested
**Test 8**|Enrure that you can distance grab|Success:
**Test 9**|Bring up and edit an objects with hand controllers|Untested
**Test 10**|Grab objects in dev-demo|undefined
**Test 11**|Test ignore function|undefined

**Comment by OmegaHeron** [comment]
"In Dev-Demo, there is an area between the whiteboard and the tether ball, that while the avatar stands in it, the sky flickers from black to the original image. Looking left and right consistantly triggers the flickering."

That's a (bad) interaction between nested zones...  Hopefully something that will be addressed someday.

**Comment by claudioshg** [comment]
Thanks for letting me know Omega. Appreciate it. 

    On Saturday, December 17, 2016 1:45 PM, Joe Large [Omega Heron] <notifications@github.com> wrote:
 

 "In Dev-Demo, there is an area between the whiteboard and the tether ball, that while the avatar stands in it, the sky flickers from black to the original image. Looking left and right consistantly triggers the flickering."That's a (bad) interaction between nested zones... Hopefully something that will be addressed someday.—

**Comment by OmegaHeron** [comment]
QA Fail

Starting with DEV-5811 PR #9213 -- Sandbox scripted agent no longer properly runs -- my (very) bad for not catching that earlier.

AC log shows following as AC script tries to connect;

`[12/18 11:52:04] [DEBUG] [hifi.networking] [1528] [avatar-mixer] Added "Agent" (I) {394383b3-2d24-48ea-b17d-af645b6eccad} 70.127.20.117:62206 / 192.168.0.6:62206 `
`[12/18 11:52:04] [DEBUG] [hifi.networking] [1528] [avatar-mixer] Activating local socket for network peer with ID "394383b3-2d24-48ea-b17d-af645b6eccad"`
`[12/18 11:52:04] [DEBUG] [hifi.networking] [1528] [avatar-mixer] udt::Socket error - QAbstractSocket::ConnectionRefusedError`
`[12/18 11:52:08] [SUPPRESS] [] [1528] [avatar-mixer] 16 repeated log entries matching "Packet of type \d+ \([\sa-zA-Z:]+\) received from unknown node with UUID" - Last entry: "Packet of type 3 (Ping) received from unknown node with UUID 394383b3-2d24-48ea-b17d-af645b6eccad"`
`[12/18 11:52:08] [SUPPRESS] [] [1528] [avatar-mixer] 142 repeated log entries matching "udt::Socket error - " - Last entry: "udt::Socket error - QAbstractSocket::ConnectionRefusedError"`
`[12/18 11:52:08] [DEBUG] [hifi.networking] [1528] [avatar-mixer] udt::Socket error - QAbstractSocket::ConnectionRefusedError`
`[12/18 11:52:12] [DEBUG] [hifi.networking] [1528] [avatar-mixer] Killed "Agent" (I) {394383b3-2d24-48ea-b17d-af645b6eccad} 70.127.20.117:62206 / 192.168.0.6:62206 `
`[12/18 11:52:13] [SUPPRESS] [] [1528] [avatar-mixer] 114 repeated log entries matching "udt::Socket error - " - Last entry: "udt::Socket error - QAbstractSocket::ConnectionRefusedError"`
`[12/18 11:52:18] [DEBUG] [hifi.networking] [1528] [avatar-mixer] Added "Agent" (I) {06cab77f-ac76-4538-8939-7ea9b9d218f0} 70.127.20.117:53639 / 192.168.0.6:53639 `
`[12/18 11:52:18] [DEBUG] [hifi.networking] [1528] [avatar-mixer] Activating local socket for network peer with ID "06cab77f-ac76-4538-8939-7ea9b9d218f0"`
`[12/18 11:52:19] [DEBUG] [hifi.networking] [1528] [avatar-mixer] udt::Socket error - QAbstractSocket::ConnectionRefusedError`
`[12/18 11:52:23] [SUPPRESS] [] [1528] [avatar-mixer] 153 repeated log entries matching "udt::Socket error - " - Last entry: "udt::Socket error - QAbstractSocket::ConnectionRefusedError"`
`[12/18 11:52:23] [DEBUG] [hifi.networking] [1528] [avatar-mixer] udt::Socket error - QAbstractSocket::ConnectionRefusedError`
`[12/18 11:52:27] [DEBUG] [hifi.networking] [1528] [avatar-mixer] Killed "Agent" (I) {06cab77f-ac76-4538-8939-7ea9b9d218f0} 70.127.20.117:53639 / 192.168.0.6:53639 `
`[12/18 11:52:28] [SUPPRESS] [] [1528] [avatar-mixer] 114 repeated log entries matching "udt::Socket error - " - Last entry: "udt::Socket error - QAbstractSocket::ConnectionRefusedError"`
`[12/18 11:52:33] [DEBUG] [hifi.networking] [1528] [avatar-mixer] Added "Agent" (I) {bf23e347-3eee-4353-a6fc-1b66f040b83c} 70.127.20.117:59825 / 192.168.0.6:59825 `
`[12/18 11:52:33] [DEBUG] [hifi.networking] [1528] [avatar-mixer] Activating local socket for network peer with ID "bf23e347-3eee-4353-a6fc-1b66f040b83c"`
`[12/18 11:52:34] [DEBUG] [hifi.networking] [1528] [avatar-mixer] udt::Socket error - QAbstractSocket::ConnectionRefusedError`
`[12/18 11:52:38] [SUPPRESS] [] [1528] [avatar-mixer] 151 repeated log entries matching "udt::Socket error - " - Last entry: "udt::Socket error - QAbstractSocket::ConnectionRefusedError"`

**Comment by OmegaHeron** [comment]
No QA Fail on AC scripted agent.

I'm an idiot - this was fixed by another PR and I was looking at yet another PR that did not include that fix thinking it was this one.  *This PR is fine*.  Deletes 10 trillion PRs laying about and goes grocery shopping.

**Comment by PhilipRosedale** [comment]
@Nex-Pro / piper - In the test with the GIF where Michelle was able to resize a tower part.... those parts are created in an unlocked state - and are able to be edited both with edit.js and by resizing... did you test with a locked object?  I just tested and I cannot resize locked objects with the grip handles, as expected.

**Comment by ZappoMan** [comment]
This sounds like a QA Fail to me. @cozza13 @RyanDowne - why are we QA Passing it.. lot's of problematic notes in here

**Comment by ZappoMan** [comment]
Windows 8 - i7 CPU - GTX980 GPU
**QA Test:** 
Overall: **QA Maybe Fail**
Notes:
- Load time for home (zen garden) content _**extremely slow.**_ ~1 min while in desktop, from load point.
- When quitting interface, a zombie process seems to hang out, preventing repeat startup (this didn't repro 100%, but did happen on first run after install)
- When switching from desktop to Oculus, the HUD bar shifted "almost out of view" (it was half in view/ha;f out of view)
- While in this shifted state, the hand lasers were unable to click on and move the HUD bar
    - I was able to move with mouse, but it was a laser fail
    - I was unable to easily reproduce the intersection failure
- In HMD (Oculus) - The red and blue lasers look good, but the purple "far grab" laser has a bad stereo mismatch and is extremely distracting.
- While near grabbing with Touch controllers, I often see the thing I'm holding sort of "snap to my hips"... this might be a hand controller tracking issue.
- After closing running scripts, I can't get it to appear again, had to restart interface to get it back. (this wasn't a 100% repro)

**Comment by cozza13** [comment]
tested on min spec AMD and Min spec AMD + Windows 7.
